apps/system/flash_test/flash_test.c

```diff
@@ -59,80 +59,570 @@
 #include <errno.h>
 #include <stdint.h>
 #include <sys/types.h>
-
+#include <sys/time.h>
+#include <arch/board/board.h>
+#include <tinyara/spi/spi.h>
 #include <tinyara/fs/mtd.h>
-
+#include <time.h>
+#include <tinyara/config.h>
+#include <tinyara/fs/ioctl.h>
+#include <fcntl.h>
 /****************************************************************************
  * Definitions
  ****************************************************************************/
-#define BUF_SIZE   1024
-
+#define BUF_SIZE   65536
+#define SIZE_1MB   1024*1024 
 #define READ_CMD    "read"
+#define WRITE_CMD   "write"
+#define ERASE_CMD   "erase"
+#define BCDEV_CMD	"bch"
+
+#ifdef CONFIG_BUILD_FLAT
+#define INT_FLASH_PART 7
+#define EXT_FLASH_PART 13
+#else
+#define INT_FLASH_PART 11
+#define EXT_FLASH_PART 17
+#endif
```

Comment:
The numbers mean?

Reply:
this is the partition number which will be used during the mtd interaction . currently i have fixed to this partition. If we want then we can get as an input from the user itself.

Reply:
updated the code to scan the specific partition ("flash_type") & use it for the testing.